tc-messaging/src/main/java/com/tc/entity/NetworkVoltronEntityMessageImpl.java

```diff
@@ -150,12 +159,21 @@ protected boolean hydrateValue(byte name) throws IOException {
     this.extendedData = getBytesArray();
     this.requiresReplication = getBooleanValue();
     this.oldestTransactionPending = new TransactionID(getLongValue());
+    
+    try {
+      if (this.type == Type.INVOKE_ACTION) {
+        MessageCodec<EntityMessage, EntityResponse> codec = supplier.getMessageCodec(this.entityDescriptor.getEntityID());
+        this.message = codec.decodeMessage(extendedData);
+      }
+    } catch (MessageCodecException exception) {
+//  swallow it
+    }
+    
     return true;
   }
 
   @Override
   public EntityMessage getEntityMessage() {
-    // Network messages don't contain an identity message.
-    return null;
+    return this.message;
```

Comment:
So, does this mean that we are piggy-backing on that "inline message reference" idea used by IEntityMessenger and RetirementManager support?  I am surprised that we ended up finding  more general-case use for what, at the time, seemed like a bit of a hack.